crates/holochain/src/core/workflow/call_zome_workflow/unit_tests.rs

```diff
@@ -0,0 +1,163 @@
+        workflow::{call_zome_workflow, CallZomeWorkflowArgs},
+    },
+    sweettest::{SweetConductor, SweetDnaFile},
+    test_utils::new_zome_call_params,
+};
+use ::fixt::fixt;
+use hdk::prelude::CellId;
+use holo_hash::fixt::ActionHashFixturator;
+use holochain_keystore::MetaLairClient;
+use holochain_p2p::{actor::MockHcP2p, HolochainP2pDna};
+use holochain_state::host_fn_workspace::SourceChainWorkspace;
+use holochain_wasm_test_utils::TestWasm;
+use kitsune2_api::DhtArc;
+use std::sync::Arc;
+
+#[tokio::test(flavor = "multi_thread")]
+async fn trigger_integration_workflow_after_creating_ops() {
+    let mut hc_p2p = MockHcP2p::new();
+    hc_p2p
+        .expect_target_arcs()
+        .returning(|_| Box::pin(async { Ok(vec![DhtArc::FULL]) }));
+    let TestCase {
+        workspace,
+        network,
+        keystore,
+        args,
+        _conductor,
+    } = TestCase::new(hc_p2p, "create", ()).await;
+    let (trigger_publish_dht_ops, _) = TriggerSender::new();
+    let (trigger_integrate_dht_ops, mut integrate_dht_ops_rx) = TriggerSender::new();
+    let (trigger_countersigning, _) = TriggerSender::new();
+
+    let _ = call_zome_workflow(
+        workspace,
+        network,
+        keystore,
+        args,
+        trigger_publish_dht_ops,
+        trigger_integrate_dht_ops,
+        trigger_countersigning,
+    )
+    .await
+    .unwrap()
+    .unwrap();
+    // Assert the integration workflow has been triggered.
+    integrate_dht_ops_rx.listen().await.unwrap();
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn integration_workflow_is_not_triggered_when_no_data_has_been_created() {
+    let mut hc_p2p = MockHcP2p::new();
+    hc_p2p
+        .expect_target_arcs()
+        .returning(|_| Box::pin(async { Ok(vec![DhtArc::FULL]) }));
+    hc_p2p
+        .expect_authority_for_hash()
+        .returning(|_, _| Box::pin(async { Ok(true) }));
+    let TestCase {
+        workspace,
+        network,
+        keystore,
+        args,
+        _conductor,
+        // Zome call to get action that does not exist.
+    } = TestCase::new(hc_p2p, "reed", fixt!(ActionHash)).await;
+    let (trigger_publish_dht_ops, _) = TriggerSender::new();
+    let (trigger_integrate_dht_ops, mut integrate_dht_ops_rx) = TriggerSender::new();
+    let (trigger_countersigning, _) = TriggerSender::new();
+
+    // Fail the test if the integration workflow has been triggered.
+    let integration_rx_task = tokio::task::spawn(async move {
+        // Expecting a channel closed error.
+        integrate_dht_ops_rx.listen().await.unwrap_err();
+    });
+    let _ = call_zome_workflow(
+        workspace,
+        network,
+        keystore,
+        args,
+        trigger_publish_dht_ops,
+        trigger_integrate_dht_ops,
+        trigger_countersigning,
+    )
+    .await
+    .unwrap()
+    .unwrap();
+    // Awaiting the task to ensure integration has not been triggered.
+    integration_rx_task.await.unwrap();
```

Comment:
I added a `try_recv` to the `TriggerReceiver` that is only available in tests. I think that's appropriate here?